main/src/modules/reactiveLayer/sensationManager/src/opcSensation.cpp

```diff
@@ -133,16 +137,78 @@ Bottle OpcSensation::handleEntities()
             else if (entity->entity_type() == "object" && dynamic_cast<Object*>(entity)->m_present == 1.0) {  // Known entities and present!
                 known_obj = true;
                 addToEntityList(k_entities, entity->entity_type(), entity->name());
+                //Handle red balls
+                if (entity->name() == "red_ball"){
+                    yDebug("Found a redBall object");
+                    if (!Network::isConnected("/pf3dTracker/data:o",("/"+moduleName+"/pf3dTracker:i").c_str())){
+                        yDebug("RedBall Port not connected...");
+                        dynamic_cast<Object*>(entity)->m_present = 0.0;
+                    }else{
+                        yDebug("RedBall Port connected...\nUpdateing position...");
+                        //update position
+                        Bottle *bot = pf3dTrackerPort.read();
+                        yDebug("RedBall Port readed");
+                        Object* obj1 = iCub->opc->addOrRetrieveEntity<Object>("red_ball");
+                        obj1->m_ego_position[0] = bot->get(0).asDouble();
+                        obj1->m_ego_position[1] = bot->get(1).asDouble();
+                        obj1->m_ego_position[2] = bot->get(2).asDouble();
+                        obj1->m_dimensions[0] = bot->get(3).asDouble();
+                        obj1->m_dimensions[1] = bot->get(4).asDouble();
+                        obj1->m_dimensions[2] = bot->get(5).asDouble();
+                        obj1->m_present = bot->get(6).asDouble();
+                        iCub->opc->commit(obj1);
+                        yDebug("opc updated...");
+                    }
+                }
+                Object* obj1 = iCub->opc->addOrRetrieveEntity<Object>(entity->name());
```

Comment:
1. Casting done. 
2. Is the first line in [here](https://github.com/robotology/wysiwyd/blob/feature/upf/value/main/src/modules/reactiveLayer/sensationManager/src/opcSensation.cpp#L137-L152) doing the same thing?